packages/cli/src/execution-lifecycle/execution-lifecycle-hooks.ts

```diff
@@ -485,13 +466,14 @@ export function getWorkflowHooksIntegrated(
 	mode: WorkflowExecuteMode,
 	executionId: string,
 	workflowData: IWorkflowBase,
+	userId: string | undefined,
 ): WorkflowHooks {
-	const hookFunctions = hookFunctionsSave();
-	const preExecuteFunctions = hookFunctionsPreExecute();
-	for (const key of Object.keys(preExecuteFunctions)) {
-		const hooks = hookFunctions[key] ?? [];
-		hooks.push.apply(hookFunctions[key], preExecuteFunctions[key]);
-	}
+	const hookFunctions = mergeHookFunctions(
+		hookFunctionsWorkflowTelemetry(userId),
+		hookFunctionsNodeTelemetry(),
```

Comment:
Now that these hooks are no longer in `WorkflowRunner` and in `WorkflowExecuteAdditionalData`, can it happen that events reach listeners in the wrong sequence? e.g. log streaming writing a node-execute-before event before writing a workflow-execute-before event. Is that a risk?

Reply:
that was what I was concerned about when I originally started this refactor in #12364 .
I've since tested this a bunch already, and haven't seen any issues so far.
also, the order of these was already inconsistent between various modes.
The only thing that I can tell is affected by these bring reordered is the final status determination. but to address that perhaps we should do that before any other `workflowExecuteAfter` hook in every mode. WDYT?

Reply:
moved `determineFinalExecutionStatus` to be before any `workflowExecuteAfter` hook that depends on `status`. now the all hooks should use the finalized status.